breadbox/breadbox/api/types.py

```diff
@@ -644,7 +689,7 @@ def delete_dimension_type_endpoint(
 
     # don't count datasets which are actually the metadata for this type
     datasets_with_using_type = [
-        x for x in datasets_with_using_type if x.id == dim_feature_type.dataset_id
+        x for x in datasets_with_using_type if x.id != dim_feature_type.dataset_id
```

Comment:
@pgm I encountered a bug which seemed related to this line while testing my new endpoint. I believe based on your comment this change should be your intended behavior. I've also added more test cases to your `test_all_dimension_type_methods` to cover what I believe this line was intending

Reply:
Yes, you're correct. This change fixes my mistake.